src/azure-cli/azure/cli/command_modules/appservice/appservice_domains.py

```diff
@@ -50,6 +50,16 @@ def create_domain(cmd, resource_group_name, hostname, contact_info, privacy=True
     web_client = web_client_factory(cmd.cli_ctx)
     hostname_availability = web_client.domains.check_availability(NameIdentifier(name=hostname))
 
+    if not hostname_availability.available:
+        raise CLIError("Custom domain name '{}' is not available. Please try again "
+                       "with a new hostname.".format(hostname))
+
+    tld = '.'.join(hostname.split('.')[1:])
+    from azure.mgmt.web.models import TopLevelDomainAgreementOption
```

Comment:
please use get_models() as 3.  in https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#multi-api-aware-modules